lib/super_digit.rb

```diff
@@ -1,15 +1,26 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(logn) n is the number itself
+# Space Complexity - O(logn)
 def super_digit(n)
-   
+    return n if n/10 == 0
+
+    sum = n % 10
+    while n/10 != 0
+        n = n/10
+        sum += n % 10
+    end
+    return super_digit(sum)
 end
-  
 
-# Time Complexity - ?
-# Space Complexity - ?
+
+# Time Complexity - O(logn) n is the number itself
+# Space Complexity - O(logn)
 def refined_super_digit(n, k)
-    
+    return n if k == 1 && n/10 == 0
+    sum = 0
+    k.times do
+        sum += super_digit(n)
```

Comment:
I suggest calling `super_digit` outside this loop, so you're not repeating the same problem over and over again.

Reply:
Oh, I see what you mean. Thanks @CheezItMan